macOS: dsymutil should not be called on static libraries

https://travis-ci.org/github/status-im/nim-beacon-chain/jobs/679826271 :

```text
error: cannot parse the debug map for "/Users/travis/build/status-im/nim-beacon-chain/build/libnfuzz.a": No such file or directory

Error: execution of an external program failed: 'dsymutil /Users/travis/build/status-im/nim-beacon-chain/build/libnfuzz.a'

make: *** [libnfuzz.a] Error 1
```

Command executed by the nim-beacon-chain Make target:

`nim c -d:release --app:staticlib --noMain --nimcache:nimcache/libnfuzz_static -o:build/libnfuzz.a  --verbosity:0 -d:usePcreHeader --passL:"-lpcre" -d:release -d:stack_size=1000000 nfuzz/libnfuzz.nim`

(there's also a "--debugger:native" in a top-level config.nims)

```
$ nim -v
Nim Compiler Version 1.3.1 [Linux: amd64]
Compiled at 2020-04-24
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: b4e9f8e814373fc38741736197d88475663ce758
active boot switches: -d:release
```

Paging @timotheecour - this probably needs an extra check for the "staticlib" target: https://github.com/nim-lang/Nim/pull/12931/files#diff-d7d89c7eb9bee7025e7b6bdc0bcd657bR842

(I see "`if optGenStaticLib in conf.globalOptions`" is available there)

--- a/compiler/extccomp.nim
+++ b/compiler/extccomp.nim
@@ -850,12 +850,12 @@ proc execLinkCmd(conf: ConfigRef; linkCmd: string) =
 
 proc maybeRunDsymutil(conf: ConfigRef; exe: AbsoluteFile) =
   when defined(osx):
-    if optCDebug notin conf.globalOptions: return
-    # if needed, add an option to skip or override location
-    let cmd = "dsymutil " & $(exe).quoteShell
-    conf.extraCmds.add cmd
-    tryExceptOSErrorMessage(conf, "invocation of dsymutil failed."):
-      execExternalProgram(conf, cmd, hintExecuting)
+    if optCDebug in conf.globalOptions and optGenStaticLib notin conf.globalOptions:
+      # if needed, add an option to skip or override location
+      let cmd = "dsymutil " & $(exe).quoteShell
+      conf.extraCmds.add cmd
+      tryExceptOSErrorMessage(conf, "invocation of dsymutil failed."):
+        execExternalProgram(conf, cmd, hintExecuting)
 
 proc execCmdsInParallel(conf: ConfigRef; cmds: seq[string]; prettyCb: proc (idx: int)) =
   let runCb = proc (idx: int, p: Process) =

